src/transcode.rs

```diff
@@ -8,16 +8,26 @@
 
 use crate::{HcHttpGatewayError, HcHttpGatewayResult};
 use base64::{prelude::BASE64_URL_SAFE, Engine};
+use holochain_client::ConductorApiError;
 use holochain_types::prelude::ExternIO;
 
-/// Function to transcode an incoming base64 encoded payload to Holochain serialized bytes
-/// (type `ExternIO`).
-pub fn base64_json_to_hsb(base64_encoded_payload: &str) -> HcHttpGatewayResult<ExternIO> {
-    let base64_decoded_payload = BASE64_URL_SAFE
-        .decode(base64_encoded_payload)
-        .map_err(|_| HcHttpGatewayError::RequestMalformed("Invalid base64 encoding".to_string()))?;
-    let json_payload = serde_json::from_slice::<serde_json::Value>(&base64_decoded_payload)
-        .map_err(|_| HcHttpGatewayError::RequestMalformed("Invalid JSON value".to_string()))?;
+/// Function to transcode an optional base64 encoded payload to Holochain serialized bytes
+/// (type `ExternIO`). If no payload is passed in, a unit value will be serialized.
+pub fn base64_json_to_hsb(
+    maybe_base64_encoded_payload: Option<String>,
+) -> HcHttpGatewayResult<ExternIO> {
+    let json_payload = if let Some(base64_encoded_payload) = maybe_base64_encoded_payload {
+        let base64_decoded_payload =
+            BASE64_URL_SAFE
+                .decode(base64_encoded_payload)
+                .map_err(|_| {
+                    HcHttpGatewayError::RequestMalformed("Invalid base64 encoding".to_string())
+                })?;
+        serde_json::from_slice::<serde_json::Value>(&base64_decoded_payload)
+            .map_err(|_| HcHttpGatewayError::RequestMalformed("Invalid JSON value".to_string()))?
+    } else {
+        serde_json::Value::Null
```

Comment:
Ahh, nice okay. I was wondering about this last night. It seems silly for people to have to encode null and pass it if they want to call a zome function that doesn't have parameters. This isn't in the spec but we should just add it.